pyout/styling.py

```diff
@@ -0,0 +1,165 @@
+                           {"type": "array",
+                            "items": [
+                                {"type": "array",
+                                 "items": [{"type": ["number", "null"]},
+                                           {"type": ["number", "null"]},
+                                           {"type": ["string", "boolean"]}],
+                                 "additionalItems": False}]}},
+            "additionalProperties": False},
+        "label": {
+            "description": "Map a value to a style",
+            "type": "object",
+            "properties": {"label": {"type": "object"}},
+            "additionalProperties": False}
+    },
+    "type": "object",
+    "properties": {
+        "default_": {
+            "description": "Default style of columns",
+            "oneOf": [{"$ref": "#/definitions/styles"},
+                      {"type": "null"}],
+            "default": {"align": "left",
+                        "width": "auto"},
+            "scope": "table"},
+        "header_": {
+            "description": "Attributes for the header row",
+            "oneOf": [{"type": "object",
+                       "properties":
+                       {"color": {"$ref": "#/definitions/color"},
+                        "bold": {"$ref": "#/definitions/bold"},
+                        "underline": {"$ref": "#/definitions/underline"}}},
+                      {"type": "null"}],
+            "default": None,
+            "scope": "table"},
+        "separator_": {
+            "description": "Separator used between fields",
+            "type": "string",
+            "default": " ",
+            "scope": "table"}
+    },
+    ## All other keys are column names.
+    "additionalProperties": {"$ref": "#/definitions/styles"}
+}
+
+
+def default(prop):
+    """Return the default value schema property.
+
+    Parameters
+    ----------
+    prop : str
+        A key for schema["properties"]
+    """
+    return schema["properties"][prop]["default"]
+
+
+def adopt(style, new_style):
+    if new_style is None:
+        return style
+
+    combined = {}
+    for key, value in style.items():
+        if isinstance(value, Mapping):
+            combined[key] = dict(value, **new_style.get(key, {}))
+        else:
+            combined[key] = new_style.get(key, value)
+    return combined
+
+
+class StyleError(Exception):
+    """Exception raised for an invalid style.
+    """
+    def __init__(self, original_exception):
+        msg = ("Invalid style\n\n{}\n\n\n"
+               "See pyout.schema for style definition."
+               .format(original_exception))
+        super(StyleError, self).__init__(msg)
+
+
+def validate(style):
+    """Check `style` against pyout.styling.schema.
+
+    Parameters
+    ----------
+    style : dict
+        Style object to validate.
+
+    Raises
+    ------
+    StyleError if `style` is not valid.
+    """
+    try:
+        import jsonschema
+    except ImportError:
```

Comment:
consider me anal, but that would be really cool to have 100% coverage ;)
you could mock import in the test and raise ImportError to verify that whatever code uses `validate` doesn't blow, and that would give coverage here.  If you are 100% that it is all working, you could add `# pragma: no cover` to the except line to avoid testing that clause

Reply:
I knew you were going to mention that.

> you could mock import in the test and raise ImportError to verify that whatever code uses validate doesn't blow

yes, I imagine you want something beautiful like [this]?

> you could add # pragma: no cover

I know, but I just don't care about 100% :)

I'll eagerly await your PR.

[this]: https://stackoverflow.com/a/7435557

Reply:
More seriously, we talked before about setting up the travis matrix so
that it has a non-jsonschema run.  I still plan to look into that.